### DISCUSSION TITLE
CSRF attacksproblem

### DISCUSSION THREAD

**Discussion by marcjordanavila** [primefaces/community#2562] [PrimeFaces] [answered]
Hi! 

Recently, my web page has passed a pentest and the company that did it has found a problem with the CSRF attacks. 

I am using:

- Primefaces 10.0.0
- Wildfly 23

I have read that Primefaces 10.0.0 has automatic XSS and XSRF prevention [primefaces documentation](https://primefaces.github.io/primefaces/10_0_0/#/?id=main) and JSF is protected too [cross-site-request-forgery-prevention-via-referer-header](https://stackoverflow.com/questions/61705713/cross-site-request-forgery-prevention-via-referer-header) and [csrf-xss-and-sql-injection-attack-prevention-in-jsf](https://stackoverflow.com/questions/7722159/csrf-xss-and-sql-injection-attack-prevention-in-jsf).

I have looked for something that validates this in the back-end request, and effectively, there are request body parameters that make you think that is preventing those attacks.

- primefaces.nonce
- javax.faces.ViewState

The problem is when I change or delete the parameter primefaces.nonce, because the request works fine too. (I have done it with Burp Suite)

I have added this, to my web.xml file:

`<context-param>
<param-name>javax.faces.ENABLE_CSRF</param-name>
	    <param-value>true</param-value>
	</context-param>
	<context-param>
	    <param-name>javax.faces.STATE_SAVING_METHOD</param-name>
	    <param-value>server</param-value>
	</context-param>`

I don't know if I am misunderstanding something. Can somebody help me?

**Comment by vitsin** [comment]
Even on PF v14 I've ended up with this code for pages **not** protected by Spring Security:
```
<form>
<input type="hidden" name="${_csrf.parameterName}" value="${_csrf.token}"/>
</form>

**Comment by jepsar** [comment, marked as answer]
If you are concerned with security, please upgrade your PrimeFaces version. There are several developers on PF which undergo pen-tests with their applications and improvements are made constantly.

**Reply by melloware** [reply]
Agree with @jepsar i have been through multiple penetration tests adn passes them all with CSP and JSF.

if you remove or manipulate `javax.faces.ViewState` you will get an error.

if you remove or manipulate `primefaces.nonce` you will get an error but that code is not in 10.0.0 i suggest upgrading to PrimeFaces 14.0.4 when possible.
